File: direct-file/fact-graph-scala/js/src/main/scala/gov/irs/factgraph/definitions/fact/OptionConfig.scala
package gov.irs.factgraph.definitions.fact

import gov.irs.factgraph.definitions.fact.{CommonOptionConfigTraits, OptionConfigTrait}

import scala.scalajs.js.annotation.{JSExport, JSExportTopLevel}
import scala.scalajs.js

case class OptionConfig(name: String, value: String) extends OptionConfigTrait

object OptionConfig {
  def create(name: String, value: String): OptionConfig =
    OptionConfig(name, value)

  def create(
      pairs: Iterable[(String, String)],
  ): scala.scalajs.js.Array[OptionConfig] =
    val array = new scala.scalajs.js.Array[OptionConfig](pairs.size)
    pairs.map(x => create(x._1, x._2)).foreach(x => array.push(x))
    array

  def path(path: String): scala.scalajs.js.Array[OptionConfig] =
    new scala.scalajs.js.Array[OptionConfig] {
      create(CommonOptionConfigTraits.PATH, path)
    }

}

File: direct-file/backend/src/main/java/gov/irs/directfile/api/factgraph/OptionConfig.java
package gov.irs.directfile.api.factgraph;

public class OptionConfig implements gov.irs.factgraph.definitions.fact.OptionConfigTrait {
    public OptionConfig(String name, String value) {
        this.name = name;
        this.value = value;
    }

    public String name;
    public String value;

    @Override
    public String name() {
        return name;
    }

    @Override
    public String value() {
        return value;
    }
}
